guide_drafts/trait_derives.md

```diff
@@ -398,80 +421,58 @@ fn main() {
 
-error[E0277]: the trait bound `&diesel_demo_cli::models::User: diesel::Identifiable` is not satisfied
-  --> src/bin/main.rs:39:5
+error[E0277]: the trait bound `diesel_demo::models::User: diesel::associations::HasTable` is not satisfied
+  --> src/bin/main.rs:37:5
    |
-39 |     diesel::update(&hero).set((
-   |     ^^^^^^^^^^^^^^ the trait `diesel::Identifiable` is not implemented for `&diesel_demo_cli::models::User`
+37 |     diesel::update(&hero).set(
+   |     ^^^^^^^^^^^^^^^^^^^^^ the trait `diesel::associations::HasTable` is not implemented for `diesel_demo::models::User`
    |
-   = note: required because of the requirements on the impl of `diesel::query_builder::IntoUpdateTarget` for `&diesel_demo_cli::models::User`
-   = note: required by `diesel::update`
+   = note: required because of the requirements on the impl of `diesel::associations::HasTable` for `&diesel_demo::models::User`
+
+continued strack track...
 ```
+The first error shows that the compiler is expecting to find a field `id` on the struct
+even though we wanted the method.
+The note associated with this error is very important,
+which tells us the `Identifiable` trait is not implemented for our `User` struct.
 
-From this error,
-you can see the compiler assumes we are looking for a field on the struct,
-but what we want is just the value of the primary key,
-which happens to also be the `id` field.
-The compiler is also giving us some hints about trait bounds in each one of those *note* sections.
-The first two notes about `HasTable` and `IntoUpdateTarget` 
-might be helpful if we refer to the [Diesel API docs][],
-but the third error lets us know exactly which trait isn't satisfied! :smile:
+The compiler is also giving us some more hints about trait bounds in each one of those *note* sections.
+The error messages regarding `IntoUpdateTarget` and `HasTable` are due to the trait bounds
+on the `update()` method that we called after `id()`.
 
 [Diesel API docs]: https://docs.diesel.rs/diesel/associations/trait.Identifiable.html
 
-The following code is generated by [`cargo expand`][] to get our example code working again.
-[`cargo expand`][] is a crate that expands the macros and derives tat rustc generates for our code.
-
-[`cargo expand`]: https://github.com/dtolnay/cargo-expand
-
-```rust
-#[derive(Queryable)]
-pub struct User {
-    pub id: i32,
-    pub first_name: String,
-    pub last_name: String,
-    pub email: String,
-}
-
-impl <'ident> Identifiable for &'ident User {
-    type Id = (&'ident i32);
-
-    fn id(self) -> Self::Id {
-        (&self.id)
-    }
-}
-
-impl HasTable for User {
-    type Table = users::table;
-
-    fn table() -> Self::Table {
-        users::table
-    }
-}
-```
-
 ## <a name="aschangeset"></a>AsChangeset
 
-For more slightly more complicated database updates,
-[`#[derive(AsChangeset)]`][aschangeset_doc] makes your code more ergonomic.
+What happens when you have a table with a larger number of fields?
```

Comment:
Should this focus on the use case that `AsChangeset` is designed for? (See the previous comment about insertable and web forms)